### ISSUE TITLE
Error: API key ID and secret is required.

### ISSUE THREAD

**Issue by angelxmoreno** [stormpath/stormpath-sdk-node#215]
I keep getting this error and I am unable to figure out why.
```
events.js:85
      throw er; // Unhandled 'error' event
            ^
Error: API key ID and secret is required.
```
This is what I have:
```js
var config = require('./config');
var stormpath = require('stormpath');
var apiKey = new stormpath.ApiKey(config.get('STORMPATH_API_KEY_ID'), config.get('STORMPATH_API_KEY_SECRET'));
//confirm we have keys:
console.log(apiKey.toString()); // id: <my actual api key>, secret: <hidden>
var client = new stormpath.Client({apiKey: apiKey});
```
What am I doing wrong?

**Comment by angelxmoreno** [comment]
I was using the tag `v0.13.0` but on `v0.12.2` it seems to work just fine. Does the readme reflect the changes made in `v0.13.0` ?

**Comment by robertjd** [comment]
I haven't been able to reproduce this problem.  There shouldn't be any required changes for the `v0.13.0` release (we intended for it to backwards-compatible).

I did notice that we had the `lodash` dependency in the wrong place, and this was causing a module not found error when installing this module from scratch.  I just fixed this problem and released it as `v0.13.1`.  Could you remove and then re-install the `stormpath` module to this latest version, to see if this fixes your problem as well?  Thanks!

**Comment by angelxmoreno** [comment]
Same results. I am on an OSX machine. Using the above code in v0.13.x i get `Error: API key ID and secret is required.` while in v0.12.x it works just fine.

**Comment by robertjd** [comment]
Could you use `npm ls` to show me your package tree?  Please paste it in a gist and share the link.  I want to see if we have any strange ordering of stormpath modules going on.  Thanks!

**Comment by angelxmoreno** [comment]
Of course I can. I'm pretty ok atm using the earlier version but It would be nice to at least document the issue. I know v0.13 is relatively new so maybe most are still using v0.12?

here is the gist: [results of `npm ls`](https://gist.github.com/angelxmoreno/bcaf66c95b401a723e94)

**Comment by typerandom** [comment]
This definitely has to do with with the new way that configuration is loaded. I.e. the new configuration expects the key as:

```js
{
    client: {
        apiKey: {
            id: "...",
            secret: "..."
        }
    }
}
```

While it currently is provided as:

```js
{
    apiKey: {
        id: "...",
        secret: "..."
    }
}
```

So the easiest way to currently fix this is by simply changing your client configuration to:

```js
var client = new stormpath.Client({
    client: {
        apiKey: {
            id: config.get('STORMPATH_API_KEY_ID'),
            secret: config.get('STORMPATH_API_KEY_SECRET')
        }
    }
});
```

This is obviously an issue with the new version (0.13.0) of the SDK. And since we of course want to be backwards-compatible I'll start working on a fix for this so that other users who still use the old way of specifying an API key won't experience this issue.

**Comment by angelxmoreno** [comment]
Awesome! glad I found it. Should I submit a PR for the readme ?

**Comment by typerandom** [comment]
@angelxmoreno Please do.

I have fixed the issue now in https://github.com/stormpath/stormpath-node-config/commit/7cd9bad8d1efe3119456ddc6f7dbf1924504955b. Will close this issue as soon as we've updated it to use the latest `stormpath-config` module version.

**Comment by CodySpring** [comment]
I apologize if this is in the wrong spot but can the documentation be updated to reflect this? Particularly http://docs.stormpath.com/nodejs/api/home#quickstart, http://docs.stormpath.com/nodejs/api/client#top and http://docs.stormpath.com/nodejs/api/apiKey#top

Following the documentation led to a couple hours of frustration and searching before finally coming across this. :/

**Comment by angelxmoreno** [comment]
sorry to hear that @CodySpring the change is super recent and I reported the bug here after my own few hours of frustration.
@typerandom from what I understand, stormpath-config now works internally and we do not have to instantiate it manually not pass it to the stormpath client?

**Comment by typerandom** [comment]
@CodySpring I really apologize for this. This was a very recent change and it's not the experience we want you to have. So I'm really thankful that you wrote about your experience here.

@angelxmoreno That is correct. All will be taken care of internally, and the experience onward should be a lot better.

It's important to note that we haven't made any release yet. But once the new things are released I will update the docs accordingly and let you know.

**Comment by typerandom** [comment]
@CodySpring @angelxmoreno I have now released a new SDK version (v0.13.2). This should resolve this error. Please try it out and let me know how it works for you.
